src/ragger/navigator/interface.py

```diff
@@ -0,0 +1,293 @@
+
+        if not snaps_golden_path.is_dir():
+            raise ValueError(f"Golden snapshots directory ({snaps_golden_path}) does not exist.")
+
+        if not snaps_tmp_path.is_dir():
+            raise ValueError(f"Temporary snapshots directory ({snaps_golden_path}) does not exist.")
+
+        for i in range(0, last_img_idx + 1):
+            golden = self._get_snap_path(snaps_golden_path, i)
+            tmp = self._get_snap_path(snaps_tmp_path, i)
+            if not screenshot_equal(str(golden), str(tmp)):
+                raise ValueError(f"Screenshots {tmp} does not match golden.")
+
+    @abstractmethod
+    def navigate(self, instructions: List[NavigationInstruction]) -> None:
+        """
+        Navigate on the device according to a set of navigation instructions provided.
+
+        :param instructions: Set of navigation instructions.
+        :type instructions: List[NavigationInstruction]
+
+        :raises NotImplementedError: If the navigation instruction is not implemented.
+
+        :return: None
+        :rtype: NoneType
+        """
+        raise NotImplementedError
+
+    def navigate_and_compare(self,
+                             path: Path,
+                             test_case_name: Path,
+                             instructions: List[NavigationInstruction],
+                             first_instruction_wait: float = 0.5,
+                             middle_instruction_wait: float = 0.1,
+                             last_instruction_wait: float = 0.5) -> None:
+        """
+        Navigate on the device according to a set of navigation instructions
+        provided then compare each step snapshot with "golden images".
+
+        :param path: Absolute path to the snapshots directory.
+        :type path: Path
+        :param test_case_name: Relative path to the test case snapshots directory (from path).
+        :type test_case_name: Path
+        :param instructions: List of navigation instructions.
+        :type instructions: List[NavigationInstruction]
+        :param first_instruction_wait: Sleeping time before the first snapshot
+        :type first_instruction_wait: float
+        :param middle_instruction_wait: Sleeping time between between each navigation instruction
+        :type middle_instruction_wait: float
+        :param last_instruction_wait: Sleeping time after the last navigation instruction
+        :type last_instruction_wait: float
+
+        :raises ValueError: If one of the snapshots do not match.
+
+        :return: None
+        :rtype: NoneType
+        """
+        if not isinstance(self._backend, SpeculosBackend):
+            # When not using Speculos backend, taking snapshots is not possible
+            # therefore comparison is not possible too and eventually this method
+            # fallback to navigate one.
+            return self.navigate(instructions)
+
+        snaps_tmp_path = self._init_snaps_temp_dir(path, test_case_name)
+
+        sleep(first_instruction_wait)
```

Comment:
Perfectly fine for a first step but it would be nice to eventually wait an event from speculos stating it is finished displaying
Maybe add a todo ?

Reply:
yeah, could be nice to catch events for this...
I will add a todo for now.